providers/dns/designate/designate.go

```diff
@@ -240,7 +240,7 @@ func (d *DNSProvider) getRecord(zoneID string, wanted string) (*recordsets.Recor
 	}
 
 	for _, record := range allRecords {
-		if record.Name == wanted {
+		if record.Name == wanted && record.Type == "TXT" {
```

Comment:
@Idez if you prefer that, I can alternatively check if the record is not one of `SOA` or `NS` if you prefer.
```suggestion
		if record.Name == wanted && record.Type != "SOA" && record.Type != "NS" {
```